File: src/main/java/com/autosale/shop/model/UserSession.java
package com.autosale.shop.model;

import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.Value;

import java.util.UUID;

@Value
public class UserSession {
    UUID id;
    User user;

    @JsonCreator
    public UserSession(@JsonProperty("id") UUID id, @JsonProperty("user") User user) {
        this.id = id;
        this.user = user;
    }
}

File: src/main/java/com/autosale/shop/model/UserSession.kt
package com.autosale.shop.model

import com.fasterxml.jackson.annotation.JsonCreator
import com.fasterxml.jackson.annotation.JsonProperty
import java.util.*

data class UserSession @JsonCreator constructor(
    @JsonProperty("id") val id: UUID,
    @JsonProperty("user") val user: User
)
